lib/enterprise-event-stream.js

```diff
@@ -0,0 +1,234 @@
+/**
+ * @fileoverview Enterprise event stream backed by the enterprise events API
+ */
+
+'use strict';
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const Readable = require('stream').Readable,
+	util = require('util');
+
+// ------------------------------------------------------------------------------
+// Private
+// ------------------------------------------------------------------------------
+
+const DEFAULT_OPTIONS = Object.freeze({
+	pollingInterval: 60,	// seconds
+	chunkSize: 500
+
+});// ------------------------------------------------------------------------------
+// Public
+// ------------------------------------------------------------------------------
+
+/**
+ * Stream of Box enterprise events.
+ *
+ * By default, the stream starts from the current time.
+ * Pass 'startDate' to start from a specific time.
+ * Pass 'streamPosition' to start from a previous stream position, or '0' for all available past events (~1 year).
+ * Once the stream catches up to the current time, it will begin polling every 'pollingInterval' seconds.
+ * If 'pollingInterval' = 0, then the stream will end when it catches up to the current time (no polling).
+ *
+ * @param {BoxClient} client - The client to use to get events
+ * @param {Object} [options] - Options
+ * @param {string} [options.streamPosition] - The stream position to start from (pass '0' for all past events)
+ * @param {string} [options.startDate] - The date to start from
+ * @param {string} [options.endDate] - The date to end at
+ * @param {EventType[]} [options.eventTypeFilter] - Array of event types to return
+ * @param {int} [options.pollingInterval=60] - Polling interval (in seconds).  Pass 0 for no polling.
```

Comment:
Unfortunately, the API doesn't return anything to tell you that you have received all events up to `endDate`, so the SDK will keep polling indefinitely in this case.  Plus, endDate could be in the future.  Perhaps we can improve this if the API gets cleaner in the future?